src/impl/tpls/KokkosBlas3_gemm_tpl_spec_decl.hpp

```diff
@@ -817,4 +380,166 @@ KOKKOSBLAS3_CGEMM_CUBLAS(Kokkos::LayoutRight, Kokkos::LayoutRight,
+                                  ROCBLAS_FN, LAYOUT, MEM_SPACE,             \
+                                  ETI_SPEC_AVAIL)                            \
+  template <class ExecSpace>                                                 \
+  struct GEMM<Kokkos::View<const SCALAR_TYPE**, LAYOUT,                      \
+                           Kokkos::Device<ExecSpace, MEM_SPACE>,             \
+                           Kokkos::MemoryTraits<Kokkos::Unmanaged> >,        \
+              Kokkos::View<const SCALAR_TYPE**, LAYOUT,                      \
+                           Kokkos::Device<ExecSpace, MEM_SPACE>,             \
+                           Kokkos::MemoryTraits<Kokkos::Unmanaged> >,        \
+              Kokkos::View<SCALAR_TYPE**, LAYOUT,                            \
+                           Kokkos::Device<ExecSpace, MEM_SPACE>,             \
+                           Kokkos::MemoryTraits<Kokkos::Unmanaged> >,        \
+              true, ETI_SPEC_AVAIL> {                                        \
+    typedef SCALAR_TYPE SCALAR;                                              \
+    typedef Kokkos::View<const SCALAR**, LAYOUT,                             \
+                         Kokkos::Device<ExecSpace, MEM_SPACE>,               \
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >           \
+        AViewType;                                                           \
+    typedef Kokkos::View<const SCALAR**, LAYOUT,                             \
+                         Kokkos::Device<ExecSpace, MEM_SPACE>,               \
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >           \
+        BViewType;                                                           \
+    typedef Kokkos::View<SCALAR**, LAYOUT,                                   \
+                         Kokkos::Device<ExecSpace, MEM_SPACE>,               \
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >           \
+        CViewType;                                                           \
+                                                                             \
+    static void gemm(const typename CViewType::execution_space& space,       \
+                     const char transA[], const char transB[],               \
+                     typename AViewType::const_value_type& alpha,            \
+                     const AViewType& A, const BViewType& B,                 \
+                     typename CViewType::const_value_type& beta,             \
+                     const CViewType& C) {                                   \
+      Kokkos::Profiling::pushRegion(                                         \
+          "KokkosBlas::gemm[TPL_ROCBLAS," #SCALAR_TYPE "]");                 \
+                                                                             \
+      KOKKOSBLAS3_GEMM_ROCBLAS_DETERMINE_ARGS(LAYOUT)                        \
```

Comment:
Unless this is re-used somewhere I would prefer remove this determine args macro.

Reply:
@e10harvey Done.